subtests/docker_cli/attach/attach.py

```diff
@@ -36,10 +36,34 @@ def wait_interactive_cmd(self):
                      % wait)
         time.sleep(wait)
 
+    def image_exist(self, image_name):
+        di = DockerImages(self)
+        image_list = DockerImages.get_dockerimages_list(di)
+        exist_status = DockerImages.filter_list_full_name(image_list,
+                                                          image_name)
+        return exist_status
+
+    def pull_image(self, image_name):
+        print "pull_image"
+        dkrcmd = AsyncDockerCmd(self, 'pull', [image_name],
+                                self.config['docker_timeout'],
+                                verbose=True)
+        self.loginfo("Executing background command: %s" % dkrcmd)
+        dkrcmd.execute()
+        while not dkrcmd.done:
+            self.loginfo("Pulling...")
+            time.sleep(3)
+        result = dkrcmd.wait()
+        self.failif(result.exit_status != 0,
+                    "Fail to download image %s"
+                    % image_name)
+
     def initialize(self):
         super(attach_base, self).initialize()
         self.sub_stuff['subargs'] = self.config['run_options_csv'].split(',')
         fin = DockerImage.full_name_from_defaults(self.config)
+        if self.image_exist(fin) == []:
+            self.pull_image(fin)
```

Comment:
The another thing i'm concern is about the `class DockerImages(object):`
1.Do we need a `def image_exist()` in the Image/Images class, it just like `def image_exist()` in my patch, to check whether an image exist? I don't see a docker command can do that directly, nor in auotest-docker library. I would write the check in the library but just in case introduce more problems, so i put it in this single case only now.

2.About the `clean_all()`,  assume there is a image full name is `luwensu.com/rhel7:latest` and i give this in `default.ini`:
`preserve_fqins = rhel7`, then `clean_all()` will remove `luwensu.com/rhel7:latest` as it thinks only `rhel7` is preserve.
I see there is a document to explain it should be the full name, but i wonder whether  we need to do more check or give more warning ?
Just not sure about it, if it deserves a talk, we can more it into a single issue.